ironfish/src/network/peers/peerManager.ts

```diff
@@ -313,16 +313,31 @@ export class PeerManager {
       { simulateLatency: this.localPeer.simulateLatency },
     )
 
-    connection.onSignal.on((data) => {
+    connection.onSignal.on(async (data) => {
+      let errorMessage
       if (peer.state.identity === null) {
-        const message = 'Cannot establish a WebRTC connection without a peer identity'
-        this.logger.debug(message)
-        connection.close(new NetworkError(message))
+        errorMessage = 'Cannot establish a WebRTC connection without a peer identity'
+      }
+
+      // Ensure one or more brokering peers exists before encrypting the signaling message,
+      // but discard the brokering peer in case its state changes during encryption
+      if (this.getBrokeringPeer(peer) == null) {
```

Comment:
It's strange that this finds a brokering peer then throws it away, only to try again down the line. The general flow here is a bit confusing, I wish we could tighten this up but I'm okay if it merges.